src/parse/parser.rs

```diff
@@ -220,12 +217,15 @@ impl<'help, 'app> Parser<'help, 'app> {
             let count = self
                 .app
                 .get_positionals()
-                .filter(|p| p.settings.is_set(ArgSettings::MultipleValues) && p.num_vals.is_none())
+                .filter(|p| {
+                    p.settings.is_set(ArgSettings::MultipleOccurrences)
+                        || (p.settings.is_set(ArgSettings::MultipleValues) && p.num_vals.is_none())
```

Comment:
Use `is_multiple()` here?

Reply:
I originall used `is_multiple` but it seemed more appropriate for the `num_vals` check to be `&&` with only `MultipleValues` since that is the only one of the two it is related to.  I'll admit, this is guesswork.